x-pack/solutions/security/plugins/security_solution/public/reports/components/ai_value/cost_savings_key_insight.tsx

```diff
@@ -0,0 +1,135 @@
+import { useFindCostSavingsPrompts } from '../../hooks/use_find_cost_savings_prompts';
+
+interface Props {
+  lensResponse: VisualizationTablesWithMeta | null;
+}
+
+export const CostSavingsKeyInsight: React.FC<Props> = ({ lensResponse }) => {
+  const {
+    euiTheme: { size },
+  } = useEuiTheme();
+
+  const { http, notifications, inference, uiSettings } = useKibana().services;
+  const connectorId = uiSettings.get<string>(DEFAULT_AI_CONNECTOR);
+  const [insightResult, setInsightResult] = useState<string>('');
+
+  const hasEnterpriseLicence = licenseService.isEnterprise();
+  const { hasAssistantPrivilege, isAssistantEnabled } = useAssistantAvailability();
+  const prompts = useFindCostSavingsPrompts({
+    context: {
+      isAssistantEnabled:
+        hasEnterpriseLicence && (isAssistantEnabled ?? false) && (hasAssistantPrivilege ?? false),
+      httpFetch: http.fetch,
+      toasts: notifications.toasts,
+    },
+  });
+  const fetchInsight = useCallback(async () => {
+    if (lensResponse && connectorId && prompts !== null) {
+      try {
+        const prompt = getPrompt(JSON.stringify(lensResponse), prompts);
+        const result = await inference.chatComplete({
+          connectorId,
+          messages: [{ role: MessageRole.User, content: prompt }],
+        });
+        setInsightResult(result.content);
+      } catch (error) {
+        setInsightResult(
+          `${i18n.INSIGHTS_ERROR} ${error.body?.message ?? error.message ?? error.toString()}`
+        );
+      }
+    }
+  }, [connectorId, lensResponse, inference, prompts]);
+
+  useEffect(() => {
+    fetchInsight();
+  }, [fetchInsight]);
+  return (
+    <div
+      data-test-subj="alertProcessingKeyInsightsContainer"
+      css={css`
+        background: linear-gradient(
+          112deg,
+          rgba(89, 159, 254, 0.08) 3.58%,
+          rgba(240, 78, 152, 0.08) 98.48%
+        );
+        border-radius: ${size.s};
+        padding: ${size.base};
+        min-height: 200px;
+      `}
+    >
+      <span>
+        <EuiFlexGroup
+          gutterSize="s"
+          alignItems="center"
+          responsive={false}
+          data-test-subj="alertProcessingKeyInsightsGreetingGroup"
+        >
+          <EuiFlexItem grow={false}>
+            <EuiIcon type="logoElastic" size="m" data-test-subj="alertProcessingKeyInsightsLogo" />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTitle size="xs">
+              <p data-test-subj="alertProcessingKeyInsightsGreeting">{i18n.KEY_INSIGHT}</p>
+            </EuiTitle>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+
+        <EuiSpacer size="m" />
+
+        <EuiText
+          size="s"
+          css={css`
+            line-height: 1.6em;
+          `}
+        >
+          {insightResult ? (
+            <div
+              // eslint-disable-next-line react/no-danger
+              dangerouslySetInnerHTML={{ __html: insightResult }}
```

Comment:
Are we ok with doing this?

Reply:
https://github.com/elastic/kibana/pull/234229/commits/ca547d1842b1c0734dab8138ce5c1a3b0dab1182

I switched to markdown as @kqualters-elastic suggested over slack, but just FYI [this still uses](https://github.com/elastic/kibana/blob/a8418d931a495f6c3081803aab1144e8634b1533/src/platform/plugins/shared/kibana_react/public/markdown/markdown.tsx#L117) `dangerouslySetInnerHTML`...